### PR TITLE
Qt6

### PR DESCRIPTION
**Pull request by jschueller**
Requires #11569, not finished

- many changes are related to QRegExp which is deprecated in favor of QRegularExpression that is already available in qt5
the problem is that the variables widgets relies on the different pattern syntaxes QRegExp offers but are not available in QRegularExpression, it would be simpler to only support perl expression (and wildcards maybe) so that everything could be moved to QRegularExpression and not duplicate some code

- many changes are related to qwt which must be bumped to 6.2

maybe I should split some of these changes in several PRs, what do you think ?

### CODE REVIEW

**Comment by adeas31** [comment]
> many changes are related to QRegExp which is deprecated in favor of QRegularExpression that is already available in qt5
the problem is that the variables widgets relies on the different pattern syntaxes QRegExp offers but are not available in QRegularExpression, it would be simpler to only support perl expression (and wildcards maybe) so that everything could be moved to QRegularExpression and not duplicate some code

I agree. The default right now is perl pattern matching syntax and I don't think people are actually using other pattern syntaxes.

> maybe I should split some of these changes in several PRs, what do you think ?

Yes please. Its too many files to review at one go. Split into several PRs and I will start reviewing them. Thanks!

File: OMEdit/OMEditLIB/MainWindow.cpp
Comment (adeas31) [inline OMEdit/OMEditLIB/MainWindow.cpp:133]: Are we need required to register our own types?

**Review by adeas31 — APPROVED** [review]
LGTM. Thanks!

**Comment by adeas31** [comment]
I will merge this one after we are done with 1.23 release. I should have reviewed it earlier but sorry I couldn't manage the time.

Regarding PatternSyntax, I believe you will make a separate PR once this is merged.

File: OMEdit/OMEditLIB/MainWindow.cpp
Comment (jschueller) [reply OMEdit/OMEditLIB/MainWindow.cpp:133]: seems that was only required for qt5

**Comment by jschueller** [comment]
OMEdit crashes in webengine libs on  ubuntu 22 (qt 6.2), but ok on arch (qt 6.7)
which qt version did you test ?

**Comment by adeas31** [comment]
```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy
```

```
qmake6 --version
QMake version 3.1
Using Qt version 6.2.4 in /usr/lib/x86_64-linux-gnu
```

I don't know how else I can test this. There is Arch WSL at the Microsoft store. I will try that.

**Comment by jschueller** [comment]
OMShell has still the issue, but OMNotebook and OMEdit are fine now.

**Comment by adeas31** [comment]
OMShell should work fine now. I also updated the Windows makefiles for OMShell to allow both qt5 and qt6.

**Comment by jschueller** [comment]
nice, I confirm OMShell works now

**Comment by adeas31** [comment]
OMNotebook is compiling and starting up fine. But it doesn't work as expected. For example, try clicking on `Hello World` example. The text block disappears instead of opening the new notebook window.

**Comment by adrpo** [comment]
It seems we cannot change Jenkinsfile in a PR opened by somebody that does not have access to the repository. 
@sjoelund any idea how to handle this beside opening a new PR and moving all the commits to it?

**Comment by adrpo** [comment]
Opened #12485 to see if we get it to work, @adeas31, feel free to put the commented things back on.

**Pull request closed without merging**
